manim/mobject/svg/text_mobject.py

```diff
@@ -1056,11 +1037,441 @@ def text2svg(self):
+        disable_ligatures: bool = False,
+        **kwargs,
+    ):
+        self.text = text
+        self.size = size
+        self.line_spacing = line_spacing
+        self.font = font
+        self.slant = slant
+        self.weight = weight
+        self.gradient = gradient
+        self.tab_width = tab_width
+
+        self.original_text = text
+        self.disable_ligatures = disable_ligatures
+        text_without_tabs = text
+        if "\t" in text:
+            text_without_tabs = text.replace("\t", " " * self.tab_width)
+
+        colormap = self.extract_color_tags()
+        gradientmap = self.extract_gradient_tags()
+
+        if self.line_spacing == -1:
+            self.line_spacing = self.size + self.size * 0.3
+        else:
+            self.line_spacing = self.size + self.size * self.line_spacing
+
+        file_name = self.text2svg()
+        PangoUtils.remove_last_M(file_name)
+        SVGMobject.__init__(
+            self,
+            file_name,
+            color=color,
+            fill_opacity=fill_opacity,
+            stroke_width=stroke_width,
+            height=height,
+            width=width,
+            should_center=should_center,
+            unpack_groups=unpack_groups,
+            **kwargs,
+        )
+        self.chars = VGroup(*self.submobjects)
+        self.text = text_without_tabs.replace(" ", "").replace("\n", "")
+
+        nppc = self.n_points_per_cubic_curve
+        for each in self:
+            if len(each.points) == 0:
+                continue
+            points = each.points
+            last = points[0]
+            each.clear_points()
+            for index, point in enumerate(points):
+                each.append_points([point])
+                if (
+                    index != len(points) - 1
+                    and (index + 1) % nppc == 0
+                    and any(point != points[index + 1])
+                ):
+                    each.add_line_to(last)
+                    last = points[index + 1]
+            each.add_line_to(last)
+
+        if self.gradient:
+            self.set_color_by_gradient(*self.gradient)
+        for col in colormap:
+            self.chars[
+                col["start"]
+                - col["start_offset"] : col["end"]
+                - col["start_offset"]
+                - col["end_offset"]
+            ].set_color(self._parse_color(col["color"]))
+        for grad in gradientmap:
+            self.chars[
+                grad["start"]
+                - grad["start_offset"] : grad["end"]
+                - grad["start_offset"]
+                - grad["end_offset"]
+            ].set_color_by_gradient(
+                *(self._parse_color(grad["from"]), self._parse_color(grad["to"]))
+            )
+        # anti-aliasing
+        if self.height is None and self.width is None:
+            self.scale(TEXT_MOB_SCALE_FACTOR)
+
+    def text2hash(self):
+        """Internally used function.
+        Generates ``sha256`` hash for file name.
```

Comment:
Same comment as the previous one.

```suggestion
        """Generates ``sha256`` hash for file name.
```